index.bs

```diff
@@ -368,6 +374,192 @@ dictionary SanitizerConfig {
+above easily extends to [=custom data attributes=] if one considers
+{{SanitizerConfig/dataAttributes}} to be an allow-list:
+
+<table class=data>
+<tr>
+ <td></td>
+ <th>global {{SanitizerConfig/attributes}} and {{SanitizerConfig/dataAttributes}} set</th>
+</tr>
+<tr>
+ <th>local {{SanitizerElementNamespaceWithAttributes/attributes}}</th>
+ <td class=long>All [=custom data attributes=] are allowed. No [=custom data attributes=] may
+   be listed in any allow-list, as that would mean a duplicate entry.</td>
+</tr>
+<tr>
+ <th>local {{SanitizerElementNamespaceWithAttributes/removeAttributes}}</th>
+ <td class=long>A [=custom data attribute=] is allowed, unless it's listed in the local
+   remove-list.
+   No [=custom data attribute=] may be listed in the global allow-list, as that would mean a
+   duplicate entry.</td>
+</tr>
+</table>
+
+Putting these rules in words:
+
+* Duplicates and interactions between global and local lists:
+  * If a global {{SanitizerConfig/attributes}} allow list exists, then all element's local lists:
+    * If a local {{SanitizerElementNamespaceWithAttributes/attributes}} allow lists exists,
+        there may be no duplicate entries between these lists.
+    * If a local {{SanitizerElementNamespaceWithAttributes/removeAttributes}} remove lists exists,
+        then all its entries must also be listed in the global {{SanitizerConfig/attributes}}
+        allow list.
+    * If {{SanitizerConfig/dataAttributes}} is true, then no [=custom data attributes=] may be
+        listed in any of the allow-lists.
+  * If a global {{SanitizerConfig/removeAttributes}} remove list exists, then:
+    * If a local {{SanitizerElementNamespaceWithAttributes/attributes}} allow lists exists,
+        there may be no duplicate entries between these lists.
+    * If a local {{SanitizerElementNamespaceWithAttributes/removeAttributes}} remove lists exists,
+        there may be no duplicate entries between these lists.
+    * {{SanitizerConfig/dataAttributes}} must be absent.
+
+
+Note that these rules put a somewhat higher burden on configuration authors, as it's on them
+to avoid configurations which might be ambiguous.
+
+<div algorithm>
+A {{SanitizerConfig}} |config| is <dfn for=SanitizerConfig>valid</dfn> if all of the following
+conditions hold:
+
+1. The |config| has either an {{SanitizerConfig/elements}} or a {{SanitizerConfig/removeElements}}
+    [=map/key=], but not both.
+1. The |config| has either an {{SanitizerConfig/attributes}} or a {{SanitizerConfig/removeAttributes}}
+    [=map/key=], but not both.
+1. Assuming that all element names have run through [=canonicalize a sanitizer element=] algorithm,
+    and all attribute names have run through [=canonicalize a sanitizer attribute=] algorithm.
+    1. If both |config|[{{SanitizerConfig/elements}}] and
+        |config|[{{SanitizerConfig/replaceWithChildrenElements}}] [=map/exist=], then
+        the [=set/difference=] of |config|[{{SanitizerConfig/elements}}] and
```

Comment:
The *intersection* of `elements` and `replaceWithChildrenElements` must be empty.

Reply:
Done.